src/date/expand-pattern.js

```diff
@@ -38,6 +39,128 @@ return function( options, cldr ) {
 		);
 	}
 
+	function getBestMatchPattern( path, skeleton ) {
+		var availableFormats, ratedFormats, format, pattern;
+
+		pattern = cldr.main([ path, skeleton ]);
+
+		if ( skeleton && !pattern ) {
+			availableFormats = cldr.main([ path ]);
+			ratedFormats = [];
+
+			for ( format in availableFormats ) {
+				ratedFormats.push({
+					format: format,
+					pattern: availableFormats[format],
+					rate: compareFormats( skeleton, format )
+				});
+			}
```

Comment:
This block seems to me a good candidate to be cached, because it's always the same results for the same locale+calendar right? I'm fine adding a `TODO` comment for now.